### PR TITLE
refactor: [M3-6494] - React Query for Linode Migrate Dialog

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

- Use React Query hooks in `<MigrateLinode />`

## Major Changes 🔄
- Remove some usages of `listToItemsByID` 🎉
- Removes some unnecessary error states that were not used (`regionError`, `acceptError`) 🗑️
- Use all React Query hooks in `<MigrateLinode />`

## Preview 📷

> **Note**: No behavior or UI should change

Screenshot 2023-05-30 at 4 59 05 PM

## How to test 🧪
1. Spin up a Linode of any kind
2. Use the action menu on the Linode Details page or Linodes Landing page to open the **`Migrate`** dialog
3. Migrate your linode to a new region
4. Verify the migration successfully happens
5. Repeat the steps above and try breaking the Migrate dialog

### CODE REVIEW

**Review by dwiley-akamai — APPROVED** [review]

File: packages/manager/src/features/Linodes/MigrateLinode/MigrateLinode.tsx [hunk 1]
```diff
@@ -153,75 +175,55 @@ const MigrateLinode = React.memo((props: Props) => {
...
+        { variant: 'success' }
+      );
+      if (hasSignedAgreement) {
+        updateAccountAgreements({
+          eu_model: true,
+          privacy_policy: true,
+        }).catch(reportAgreementSigningError);
+      }
+      onClose();
+    });
```
Comment (abailly-akamai) [inline packages/manager/src/features/Linodes/MigrateLinode/MigrateLinode.tsx:207]: Any reason why we are losing the catch block here?

File: packages/manager/src/features/TopMenu/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -138,9 +136,11 @@ export const SearchBar: React.FC<CombinedProps> = (props) => {
...
   const types = extendTypesQueryResult(typesQuery);
 
-  const searchableLinodes = linodes.map((linode) =>
-    formatLinode(linode, types, listToItemsByID(publicImages))
-  );
+  const searchableLinodes = linodes.map((linode) => {
+    const image = _publicImages?.find((image) => image.id === linode.image);
+    const imageLabel = image?.label ?? linode.image ?? 'Unknown Image';
+    return formatLinode(linode, types, imageLabel);
+  });
```
Comment (abailly-akamai) [inline packages/manager/src/features/TopMenu/SearchBar/SearchBar.tsx:143]: Would be nice to move this to a util since it is being re-used

**Review by abailly-akamai — COMMENTED** [review]
Looks great! couple comments

Also, unrelated to your PR since it appears to be the same in PROD, but should the icon be a flashing warning one when migration is in progress? Right now it just stays gray

Screenshot 2023-06-06 at 3 48 04 PM

File: packages/manager/src/features/Linodes/MigrateLinode/MigrateLinode.tsx (hunk 1 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/Linodes/MigrateLinode/MigrateLinode.tsx:207]: React Query tracks the error and the loading state so there is nothing that needs to be done if an error happens here.

Comment (abailly-akamai) [reply packages/manager/src/features/Linodes/MigrateLinode/MigrateLinode.tsx:207]: ah great to know 👍

**Pull request merged**